### ISSUE TITLE
Pagination (suggestion)

### ISSUE THREAD

**Issue by ravi-teja-nani** [Buzzpy/Dev-Encyclopedia#15]
If the number of cards increases no need to render all cards at once, instead you can only show few cards in view and remaining in filters, search, pagination.

**Comment by Himanshuch8055** [comment]
This is a good suggestion. I think we need to consider implementing it.

**Comment by RayMathew** [comment]
How might you implement this, with the owner's objective of keeping all cards loaded in the frontend itself, and not using a DB?

**Comment by ravi-teja-nani** [comment]
> How might you implement this, with the owner's objective of keeping all cards loaded in the frontend itself, and not using a DB?

Yes, we know that the owner won't use db for now. For performance we can consider visualisation. We can keep data in list format. And render the next set of cards only when needed.

Assume you are implementing search functionality, when you find a match then you can dynamically construct cards.

**Comment by Buzzpy** [comment]
Hello there!
Now that the user base is growing and this project needs to add more terms and concepts to the list, I think it'd be a good idea to switch to a database. 

Would this increase the complexity of the project? Or, should we create a JSON file with a list of terms and automatically create cards in HTML?

**Comment by Buzzpy** [comment]
I'm currently testing the Firebase database for the project, and will keep you updated.

**Comment by bartosz-skejcik** [comment]
> How might you implement this, with the owner's objective of keeping all cards loaded in the frontend itself, and not using a DB?

We can use markdown files :D

**Comment by Rodelph** [comment]
Hello everyone. I hope that you are doing well. I have managed to implement the autocomplete feature and it looks like this. 

![Recording 2024-08-16 at 01 37 42](https://github.com/user-attachments/assets/e2e1011b-63d0-442a-883c-9e6bdc6cf906)

I did have to change quite few things in order to do this, but first let me tell you first about my approach. 

![image](https://github.com/user-attachments/assets/05ab3617-ebea-4063-9a9c-60dbcb3b0ab6)

I am currently using [Aiven](https://aiven.io/) which is a data platform that can host online mysql databases (for free to a certain degree). Since our goal for now is just autocomplete. The database will not be big, thus we don't need to pay extra for a bigger storage. The free version is 5gb which is more than enough for this project i believe and we can even scale this platform using the free version. 

For the project in question i had to turn the project into a ![Node.js](https://nodejs.org/en/about) web-app. 

![image](https://github.com/user-attachments/assets/9a5311bb-09a2-43c4-9520-317db634bfc2)

I believe doing it like this will only help us scale the project and allow us to do more things. Such as rewriting the cards for a better code structure. 

I have a question regarding the autocomplete though. Do we get rid of the previous search that existed and uniquely use the drop down version that we are working on, or it is going to be both ?

Looking forward to hear back from you, and I remain at your disposal for any further information.

**Comment by Rodelph** [comment]
Hey guys, regarding the usage of the DB can you look at my approach and give me your oppinion ?
https://github.com/Buzzpy/Dev-Encyclopedia/issues/9#issuecomment-2292467027

**Comment by Ibrahim-Faisal15** [comment]
I submitted my PR regarding pagination.

**Comment by RayMathew** [comment]
Just a heads up, any PR should wait until https://github.com/Buzzpy/Dev-Encyclopedia/issues/38 is resolved.

**Comment by Ibrahim-Faisal15** [comment]
Can you guys have a peek at my PR.

**Comment by Himanshuch8055** [comment]
@Ibrahim-Faisal15, the issue with the switch to a content-driven framework has been resolved in ticket #28. Do not create a new ticket until the issue with the switch to a content-driven framework is resolved in ticket #38.

**Comment by Sudharshaun** [comment]
Hey people! Is anyone working on the pagination support? I think it's high time we look into that side as we have around **254** cards in the DOM getting rendered. As the terms increase, the number of DOM elements that gets loading in the single go is going to get increased. 

> I'm currently testing the Firebase database for the project, and will keep you updated.

As @Buzzpy mentioned are we going towards firebase or should we try pagination in front end itself. I can do a POC if we are open to doing pagination on FE.

**Comment by Buzzpy** [comment]
The firebase idea was dismissed due to a few reasons, I'm sorry I forgot to update this 😄

I think we should try pagination in the frontend itself. Would love your support here, @Sudharshaun ! 🤍

**Comment by bellodamilola** [comment]
I can pick this up @Sudharshaun @Buzzpy

**Comment by ManorSailor** [comment]
@bellodamilola & @Sudharshaun
Are you guys working on it? I'd like to help too.

**Comment by bellodamilola** [comment]
I've started working on this already. I'm creating a solution for both a pagination case and a lazy loading infinite scroll case. I'll spend more time on this during the weekend tho. 

@ManorSailor you can try any other alternative, and then we pick which is a better solution. What do you think?

**Comment by ManorSailor** [comment]
> I've started working on this already. I'm creating a solution for both a pagination case and a lazy loading infinite scroll case. I'll spend more time on this during the weekend tho.
> 
> @ManorSailor you can try any other alternative, and then we pick which is a better solution. What do you think?

Sounds great. I am willing to go give this a try. I'll be going Astro only route ;)

**Comment by Buzzpy** [comment]
Hello there, @bellodamilola and @ManorSailor!
How's it going? Did you have time to try it out?

(No rush, just checking in 🐳)

**Comment by ManorSailor** [comment]
@Buzzpy Hey! Sorry for no updates, I have been busy with work lately. Thus, I haven't had the time to work on it yet. I will work on it soon & open a PR :)
